internal/services/apimanagement/custompollers/api_management_api_poller.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package custompollers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/api"
+	"github.com/hashicorp/go-azure-sdk/sdk/client"
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+	"github.com/hashicorp/go-azure-sdk/sdk/odata"
+)
+
+var _ pollers.PollerType = &apiManagementAPIPoller{}
+
+type apiManagementAPIPoller struct {
+	client  *api.ApiClient
+	id      api.ApiId
+	asyncID string
+}
+
+var (
+	pollingSuccess = pollers.PollResult{
+		Status:       pollers.PollingStatusSucceeded,
+		PollInterval: 10 * time.Second,
+	}
+	pollingInProgress = pollers.PollResult{
+		Status:       pollers.PollingStatusInProgress,
+		PollInterval: 10 * time.Second,
+	}
+)
+
```

Comment:
Can we add godoc here to explain the purpose of this custom polller? something like:
```suggestion
// NewAPIManagementAPIPoller - creates a new custom poller for API Management API to handle the case where... [details here]
```